server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -403,16 +407,23 @@ private CompletionStage<Result> renderErrorOrRedirectToNextBlock(
     if (featureFlags.isProgramEligibilityConditionsEnabled(request)
         && !roApplicantProgramService.isBlockEligible(blockId)) {
       CiviFormProfile submittingProfile = profileUtils.currentUserProfile(request).orElseThrow();
-      return supplyAsync(
-          () ->
-              ok(
-                  ineligibleBlockView.render(
-                      request,
-                      submittingProfile,
-                      roApplicantProgramService,
-                      applicantName,
-                      messagesApi.preferred(request),
-                      applicantId)));
+      try {
+        String externalLink = programService.getProgramDefinition(programId).externalLink();
```

Comment:
**suggestion:** Thoughts on passing the full ProgramDefinition here and below instead of the externalLink, so we're not passing a primitive and the view can benefit from having the program definition if needed for other data in the future?

Reply:
I did think about it, but I stuck with less is more. I am ok with either approach.

Reply:
@dkatzz  - I have changed per your feedback

Reply:
Thanks!